10_rest/mohitnag/pkg/rest/rest.go

```diff
@@ -0,0 +1,130 @@
+package rest
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy"
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy/store"
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/go-chi/render"
+)
+
+type Handler struct {
+	storer store.Storer
+	router chi.Router
+}
+
+func (rh *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	rh.router.ServeHTTP(w, r)
+}
+
+func NewRestHandler(s store.Storer) http.Handler {
+	r := chi.NewRouter()
+	r.Use(middleware.RequestID)
+	r.Use(middleware.Logger)
+	r.Use(middleware.Recoverer)
+	r.Use(middleware.URLFormat)
+	r.Use(render.SetContentType(render.ContentTypeJSON))
+	rh := Handler{
+		storer: s,
+		router: r}
+	rh.handleRequests()
+	return &rh
+}
+
+func (rh *Handler) handleRequests() {
```

Comment:
This name doesn't quite match what the function does. This function does not actually handle requests, it sets up a bunch of routes to tell the router how to route requests to actual handlers

Reply:
Renamed to `handleRoutes`.